### DISCUSSION TITLE
Sharing the HTML slides

### DISCUSSION THREAD

**Discussion by torannn** [jeertmans/manim-slides#120] [Q&A] [answered]
I have converted the slides to .html format. The result .html file executes normally on my PC but when I send that file to my friends, they can't run it. How can I share it with my friends? Thank you.

**Comment by jeertmans** [comment, marked as answer]
Hi @torannn, thanks for asking :-)

Maybe this should be better documented, but here is how you can share an HTML presentation:

1. Let's say you created a presentation that uses `BasicExample` from [`example.py`](https://github.com/jeertmans/manim-slides/tree/main) https://github.com/jeertmans/manim-slides/blob/35195f89e4bf3d01245e5d0351fa137cb180701f/example.py#L19-L32
   You should have first rendered the animation with something like `manim example.py BasicExample`.
2. Then, when converting to HTML, you chose a destination name, e.g., `basic.html`: `manim-slides convert BasicExample basic.html.
3. If you open the HTML file, e.g., with `--open`, you should see the videos. Note that the videos are not contained in `basic.html`, but rather in the `basic_assets/` folder. If you want to share your presentation with others, make sure to also share the `basic_assets/` folder. It should also be in the same directory as the one of `basic.html`.

I hope I answered your question :-)

**Reply by torannn** [reply]
Hi. I did like what you said and I managed to share some basic examples. I tested all the examples in your document (the circle one, the 3D one, and the big tutorial one). The .html and the correspondent assets folder of that three examples work well together on the person I shared with. 

However, when I try the same method with my own example, it just doesn't work right. It only showed the final frames while the first frame was left with a dark blank.

It would be great if you have a glance at my code, compile it on your PC, copy the assets with the html into another place and run the html. Thank you so much.
[test2.zip](https://github.com/jeertmans/manim-slides/files/10558530/test2.zip)

**Reply by jeertmans** [reply]
Hi, please share code using triple backstick syntax on Github, and not via zip :-)

For example
````raw
```python
from manim import *
# or: from manimlib import *
from manim_slides import Slide

class Test(Slide):
    def tinywait(self):
        self.wait(0.1)

    def construct(self):
        circle = Circle(radius=3, color=BLUE)
        dot = Dot()
        text = Text("Xin chào. Đây là một sản phẩm mới từ Đậu Hoàng", font="sans-serif").move_to(circle.get_center()).scale(0.7)
        text2 = Text("Điều gì sắp diễn ra?",font="sans-serif").next_to(text, DOWN).scale(0.7)
        text3 = Text("Hãy sử dụng các phím mũi tên hoặc nút Page Up/Page Down trên bàn phím để khám phá!", font="sans-serif").next_to(text2, DOWN).scale(0.7)
        text4 = Text("BOOOM", font="sans-serif").next_to(text2, DOWN).scale(2)
        self.play(GrowFromCenter(circle))
        self.tinywait()  # Waits user to press continue to go to the next slide
        self.play(Transform(circle,text))
        self.tinywait()
        self.play(SpinInFromNothing(text2))
        self.tinywait()
        self.play(SpinInFromNothing(text3))
        self.pause()
        self.play(FadeOut(text3, shift = RIGHT))
        self.play(FadeOut(text2, shift = LEFT))
        self.start_loop()  # Start loop
        self.play(FadeIn(text4), shift = DOWN)
        self.end_loop()  # This will loop until user inputs a key
        self.tinywait()
``` 
````

which will be displayed as

```python
from manim import *
# or: from manimlib import *
from manim_slides import Slide

class Test(Slide):
    def tinywait(self):
        self.wait(0.1)

    def construct(self):
        circle = Circle(radius=3, color=BLUE)
        dot = Dot()
        text = Text("Xin chào. Đây là một sản phẩm mới từ Đậu Hoàng", font="sans-serif").move_to(circle.get_center()).scale(0.7)
        text2 = Text("Điều gì sắp diễn ra?",font="sans-serif").next_to(text, DOWN).scale(0.7)
        text3 = Text("Hãy sử dụng các phím mũi tên hoặc nút Page Up/Page Down trên bàn phím để khám phá!", font="sans-serif").next_to(text2, DOWN).scale(0.7)
        text4 = Text("BOOOM", font="sans-serif").next_to(text2, DOWN).scale(2)
        self.play(GrowFromCenter(circle))
        self.tinywait()  # Waits user to press continue to go to the next slide
        self.play(Transform(circle,text))
        self.tinywait()
        self.play(SpinInFromNothing(text2))
        self.tinywait()
        self.play(SpinInFromNothing(text3))
        self.pause()
        self.play(FadeOut(text3, shift = RIGHT))
        self.play(FadeOut(text2, shift = LEFT))
        self.start_loop()  # Start loop
        self.play(FadeIn(text4), shift = DOWN)
        self.end_loop()  # This will loop until user inputs a key
        self.tinywait()
```

Anyway, regarding your issue, moving the HTML files and its assets' folder to another place works fine.

**Reply by torannn** [reply]
Thank you for pointing out that. I don't use git much 😄 . In my assets folder, there are enough partial movies, and the command `manim-slides Test` ran normally in the offline mode. But when I convert and open the .html, it doesn't show the appearance of the circle. Can you send me the folder you compiled? Or you can try running my files. Whatever I try, my device doesn't show the correct result.
[test.zip](https://github.com/jeertmans/manim-slides/files/10564248/test.zip)

**Reply by jeertmans** [reply]
There you go: [test.zip](https://github.com/jeertmans/manim-slides/files/10567182/test.zip).


If you think that some animation is missing, can you open the developer tool tab in your browser? This is usually accessed by pressed F12 on your keyboard. If an error occurs, it should be printed in the console:
![image](https://user-images.githubusercontent.com/27275099/216280031-91bb7bfa-9f76-4940-a149-2695bfc426be.png)

**Reply by jeertmans** [reply]
Ok so I compared my HTML (left) with yours (right), and they differ in the animations, see [here](https://www.diffchecker.com/tlzMFXd3/). Are you sure you have rendered your slides with manim, using `manim test.py Test` (or similar), with the exact file described above?

If so, can you share the output of the `manim test.py Test` command, as well as the output of `manim-slides convert -v DEBUG Test test.html`? Thanks!

**Reply by torannn** [reply]
Hi. So I make a new folder and I compile test.py there with the two following commands: `manim test.py Test` and `manim-slides convert -v DEBUG Test test.html`. I've included the full log file. 

In the recent testing, I was able to compile `manim test.py Test`, but I cannot convert to html (see the details error log in the file). The offline presentation `manim-slides Test` ran normally. Eventually, I ran into that error converting to html sometimes before, but then it just work fine.  If you don't mind, you can have a look on my PC (by remoting control like UltraViewer or something?). In the case you have time to look at it, we can DM later in Discord.
[Log File.txt](https://github.com/jeertmans/manim-slides/files/10580513/Log.File.txt)

**Reply by jeertmans** [reply]
Sorry for the delay @torannn, I totally forgot to reply :-/

Reading the logs, it seems that it may be due to a Windows-path-related issue. I think I can fix that easily.

**Reply by jeertmans** [reply]
Hello @torannn, would you mind testing again, but with this new version:
```
pip install git+https://github.com/jeertmans/manim-slides.git@filepath
```
? Thanks!

**Reply by torannn** [reply]
Hi @jeertmans, now the file is working perfect. Thank you for your support to Window users 😄

**Reply by jeertmans** [reply]
Thanks for the update @torannn, it will be soon available with `manim-slides==4.9.1`.

**Reply by jeertmans** [reply]
You also might be interested in the new documentation page I wrote: https://eertmans.be/manim-slides/reference/sharing.html

**Reply by torannn** [reply]
This is a great tool for math teachers to deliver great calculus lessons. The utilities of the project are now enough to make a presentation for my students 😄 . Keep up the good work.

**Reply by jeertmans** [reply]
Many thanks :D If you make your content publicly available somehow (e.g., YouTube video, website page, GitHub repo, etc.), do not hesitate to share here! I'd be happy to see what users do with Manim Slides, and I might create a gallery one day :-)
